### DISCUSSION TITLE
How do I read in a text file with a loop, line by line and CreateSeuratObject

### DISCUSSION THREAD

**Discussion by DRSEI** [satijalab/seurat#5552] [Q&A] [answered]
Hello Everyone, 
So what I wanna do is create a string array, use a loop read in each text file, and CreateSeuratObject without reading one by one. That's what I tried to do here, I am unable to make it work for txt, CSV, and matrix files from this script.
Therefore what I really would like to know is how I can get my code to work without error,  even if there is a much smarter solution.

### read many 10X and create objectes and merge:
file_list <- list.files(path = "/lymphocytes/CRC_Evan_Newell/", full.names = TRUE) #where list.files points towards a dir containing 'x number of' .txt files
names(file_list) <- basename(file_list) 

ldf <- list() #create an empty list
for (file in names(file_list)){
  seurat_data <- read.csv("seuratojct",sep = ",",
                                              header = TRUE, row.names = 1)
  ldf[[file]] <- CreateSeuratObject(counts = seurat_data, 
                                   min.features = 100, 
                                   project = file)
  assign(file, ldf[[file]])
}

SEI <- merge(ldf[[1]], ldf[2:length(ldf)]) #merging all seurat objects in list seu

dim(SEI)
table(SEI@meta.data$orig.ident)

I really appreciate your helps

**Comment by samuel-marsh** [comment, marked as answer]
Hi,

Not a Seurat answer but you could try using function fro my scCustomize package.  It provides function which does the looping for you and can even read all files in given directory just by pointing function there, you can specify specific file names but don't need to.

You can use `Read_GEO_Delim` (see link [here](https://samuel-marsh.github.io/scCustomize/articles/Read_and_Write_Functions.html#import-delimited-matrices-single-directory-with-file-prefixes) for tutorial example).  This will create a list of sparse matrices.  You can then use `Merge_Sparse_Data_All` with the `add_cell_ids` parameter (see [here](https://samuel-marsh.github.io/scCustomize/articles/Read_and_Write_Functions.html#add-barcode-prefixsuffix)) to create single matrix with barcodes from different original samples given their own prefix so that when you create Seurat object you can specify names delimiter and assign an original identiy to each object based on prefix:
`CreateSeuratObject(..., names.delim = "_", names.field = 1)`

Best,
Sam

**Reply by DRSEI** [reply]
Thank you sam
